### PR TITLE
Added color options

### PR DESCRIPTION
**Pull request by Grovkillen**
Added color option to "added", "removed" and "mark-differences". Also made it possible to blur "added" and "removed".

PS. sorry for my ignorance earlier today.

### CODE REVIEW

**Comment by vslavik** [comment]
Thanks, but can you please follow best practices, i.e. do one commit per one thing (with [good commit messages](https://cbea.ms/git-commit/)) instead of mixing completely unrelated things together? Here it's made worse by also mixing in spurious formatting and whitespace changes that make it hard to see the *actual* changes. 

Would be nice to understand the reasoning behind adding blur, which at first glance makes the diff *worse*...

File: diff-pdf.cpp [hunk 1]
```diff
@@ -170,91 +245,97 @@ cairo_surface_t *diff_images(int page, cairo_surface_t *s1, cairo_surface_t *s2,
...
-        {
-            memcpy(out, data1, r1.width * 4);
-        }
-    }
+	if (s1) {
+		unsigned char *out = datadiff + (r1.y - rdiff.y) * stridediff + (r1.x - rdiff.x) * 4;
+		for (int y = 0; y < r1.height; y++, data1 += stride1, out += stridediff) {
+			memcpy(out, data1, r1.width * 4);
+		}
+	}
```
Comment (vslavik) [inline diff-pdf.cpp:253]: Please look at this snippet as representative of the same changes all over the PR. It

- changes whitespace (probably to TABs which aren't used anywhere in the file)
- it radically reformats the code block pointlessly
- it doesn't respect the file's existing formatting (w.r.t. `{` in particular)

Please avoid making such changes and please avoid TABs. 

P.S. You can use `git push —force` to replace commits with new ones.

**Comment by BFjimmy** [comment]
Thanks for the feedback. The blur is to make things like this:

image
image

It made it, in my opinion, easier to read if changes are put on top of each other. Just an option.

**Comment by BFjimmy** [comment]
The other way around is probably better to explain my idea with the blur:
image

**Comment by vslavik** [comment]
> The other way around is probably better to explain my idea with the blur:

Sorry but neither image explains anything. Please use words? And maybe before/after images?

> It made it, in my opinion, easier to read if changes are put on top of each other. Just an option.

It's not "just an option" - each extra setting means doubling possible configurations, extra code that makes it harder to maintain, make changes etc. I'm happy to include options *if I understand their purpose*, but less so if any questions are dismissed with "it's just an option".

This also shows the important of having one commit per one thing: the 2nd thing could already be merged (after cleanup).

**Comment by Grovkillen** [comment]
Yeah, I guess I'm not cut out for meeting the standard. I'm not too good with words. We might just close this one. Sorry for messing around.

**Comment by vslavik** [comment]
> I'm not too good with words

You're not good with being good open source citizen. 

It is incredibly rude, disrespectful and inconsiderate  to ask others to waste time reviewing something only to abandon it. People have other things to do than to waste their time on your whims, you know.

**Pull request closed without merging**